lib/NormalModule.js

```diff
@@ -284,15 +284,22 @@ class NormalModule extends Module {
 				return callback();
 			}
 
+			const isMJS = this._source._name && /.mjs$/.test(this._source._name);
```

Comment:
`/.mjs$/` should be ` /\.mjs$/` 😁 

@sokra is there a common `name` property on all possible sources? Accessing the private `_name` property is not ideal here.